crates/chain/src/keychain.rs

```diff
@@ -49,6 +49,11 @@ impl Balance {
     }
 }
 
+/// A tuple of keychain index and corresponding [`ScriptBuf`].
+pub type Indexed<T> = (u32, T);
```

Comment:
maybe something like

```suggestion
/// A tuple of keychain index and an arbitrary `T` representing the indexed value.
pub type Indexed<T> = (u32, T);
```